qiskit/ignis/measurement/discriminator/iq_discriminators.py

```diff
@@ -24,6 +24,10 @@
 from qiskit.pulse import PulseError
 from qiskit.result import Result
 from qiskit.pulse.schedule import Schedule
+from qiskit.visualization import HAS_MATPLOTLIB
```

Comment:
I think I would just replace this with this code block
```
try:
    from matplotlib import pyplot as plt
    HAS_MATPLOTLIB = True
except ImportError:
    HAS_MATPLOTLIB = False
```

Because this unnecessarily runs a lot of imports in the `__init__` of the visualization class